ee/clickhouse/models/property.py

```diff
@@ -230,6 +237,38 @@ def prop_filter_json_extract(
             ),
             params,
         )
+    elif operator == "is_date_after":
+        # introducing duplication in these branches now rather than refactor too early
+        prop_value_param_key = "v{}_{}".format(prepend, idx)
+        query = f"AND parseDateTimeBestEffortOrNull({property_expr}) > %({prop_value_param_key})s"
+
+        if (
+            prop.property_definition is not None
+            and prop.property_definition.format is not None
+            and prop.property_definition.format == UNIX_TIMESTAMP_IN_SECONDS
+        ):
+            query = f"AND parseDateTimeBestEffortOrNull(substring({property_expr}, 1, 10)) > %({prop_value_param_key})s"
```

Comment:
> What happens if the datetime string has 10+ characters like 2021-04-01 18:30:00

parseDateTimeBestEffortOrNull matches a range of formats or returns null. If someone sends a datetime that ClickHouse can't parse it isn't matched in the results (because null is neither greater than nor less than any date)

> What happens if property_expr results in a non-string? What happens if the value has leading/trailing whitespace?

```
select parseDateTimeBestEffortOrNull('   2021-12-25 00:00:01')
 -- ✅ 2021-12-25 00:00:01

select parseDateTimeBestEffortOrNull('   2021-12-25 00:00:01   ')
 -- ✅ 2021-12-25 00:00:01

select parseDateTimeBestEffortOrNull('   not a date   ')
 -- ✅ null

select toDateTimeOrNull('   2021-12-25 00:00:01')
 -- ❌ null

select toDateTimeOrNull('   2021-12-25 00:00:01   ')
 -- ❌ null
```

At the point there are enough examples in the code to refactor this. I'll include a call-out to the parseDateTime docs. Since this enough is only in my head right now